reciprocalspaceship/dataset.py

```diff
@@ -17,6 +17,53 @@
     compute_structurefactor_multiplicity,
     bin_by_percentile
 )
+from functools import wraps
+from inspect import signature
+
+
+def inplace(f):
+    """ 
+    A decorator that applies the inplace argument. Base function must have a Bool param called "inplace".
+    The position of `inplace` doesn't matter.
+    """
+    @wraps(f)
+    def wrapped(ds, *args, **kwargs):
+        sig = signature(f)
+        bargs = sig.bind(ds, *args, **kwargs)
+        bargs.apply_defaults()
+        if 'inplace' in bargs.arguments:
+            if bargs.arguments['inplace']:
+                return f(ds, *args, **kwargs)
+            else:
+                return f(ds.copy(), *args, **kwargs)
+        else:
+            raise KeyError(f'"inplace" not found in local variables of @inplacemethod decorated function {f} '
+                             "Edit your method definition to include `inplace=Bool`. "
+                             )
+    return wrapped
+
+def _index_from_names(names, ds):
```

Comment:
Fine as is, but this could also become a private method of `DataSet` if we would like. Then the call signature would be `def _index_from_names(self, names)`, which could be cleaner. I'm fine with it as is, too.